docs/docs/12-context.md

```diff
@@ -123,6 +123,43 @@ By passing down the relevant info in the `Menu` component, each `MenuItem` can c
 
 Recall that you can also pass entire React components in props if you'd like to.
 
+## Referencing context in lifecycle methods
+
+If `contextTypes` is defined within a component, the following lifecycle methods will receive an additional parameter, the `context` object:
+
+```javascript
+void componentWillReceiveProps(
+  object nextProps, object nextContext
+)
+
+void shouldComponentUpdate(
+  object nextProps, object nextState, object nextContext
+)
+
+void componentWillUpdate(
+  object nextProps, object nextState, object nextContext
+)
+
+void componentDidUpdate(
+  object prevProps, object prevState, object prevContext
+)
+```
+
+## Referencing context in stateless components
+
+Stateless functional components are also able to reference `context` if `contextTypes` is defined as a property of the function. The following code shows the `Button` component above written as a stateless component.
```

Comment:
I'm fine with either "stateless functional component" or "functional component". We don't want to say "stateless component" because class-based components can be stateless (whereas they are imperative and necessarily have a backing instance; they are not functional so "functional" is the distinguishing adjective here).